src/Host/Broker/Impl/Security/SecurityManager.cs

```diff
@@ -48,6 +49,34 @@ public class SecurityManager {
             return new ClaimsPrincipal(identity);
         }
 
+        private uint CreateProfile(string username, string domain, string password,out StringBuilder profilePath, out bool profileExists) {
```

Comment:
This method is synchronous and blocking on all pipe operations. If the profile creation service hangs or goes down with the pipe still open, any request that hits this is going to get stuck, and will take down one worker thread with it.

I think it might be worthwhile to redo this in async, and use a cancellation token with a timeout on all operations. 

The catch is that binary readers and writers don't support async. So instead of using them, it might be better to aggregate the entirety of the request in a single object, then serialize the object to a string or blob, and dump that into the pipe with a single `WriteAsync` - and similarly read and process it from the other end.